src/python/bindings.cpp

```diff
@@ -902,4 +909,180 @@ PYBIND11_MODULE(pyqmap, m, py::mod_gil_not_used()) {
+position. The number of stages is specified by `num_stages`. The number of
+transfers is fixed by `num_transfers` if give. If this parameter is not
+specified, then the solver will determine the optimal number of transfers. The
+parameter `mind_ops_order` specifies whether the order of the operations in the
+circuit should be preserved. The parameter `shield_idle_qubits` specifies
+whether idle qubits should be shielded from the entangling operations.
+
+.. note::
+    To retrieve the list of qubit pairs from a quantum circuit, use the function
+    :func:`get_ops_for_solver`.
+
+.. note::
+    The returned solver's result can either directly exported to the YAML format
+    by calling the method :func:`yaml` on the result object or the result object
+    can be passed to the function :func:`generate_code` to generate code
+    consisting of neutral atom operations.
+
+:param ops: is the list of operations in the circuit
+:param num_qubits: is the number of qubits in the circuit
+:param num_stages: is the number of stages in the sequence
+:param num_transfers: (optional) is the number of transfers in the sequence
+:param mind_ops_order: is True if the order of the operations should be
+preserved
+:param shield_idle_qubits: is True if idle qubits should be shielded
+:returns: the result of the solver
+)");
+
+  py::class_<na::NASolver::Result>(
+      m, "NAStatePreparationSolver.Result",
+      "Neutral Atom State Preparation Solver Result")
+      .def(py::init<>(), "Create a result object")
+      .def(
+          "yaml",
+          [](const na::NASolver::Result& result, const bool compact) {
+            return result.yaml(0, compact);
+          },
+          "compact"_a = true, R"(
+Returns the result as a YAML string.
+
+:param compact: if True, the YAML string contains JSON elements to make it more
+compact
+:returns: the result as a YAML string
+)");
+
+  m.def(
+      "generate_code",
+      [](const py::object& circ, const na::NASolver::Result& result,
+         uint16_t maxHOffset, uint16_t maxVOffset, uint16_t minEntanglingY,
+         uint16_t maxEntanglingY, uint16_t minAtomDist,
+         uint16_t noInteractionRadius, uint16_t zoneDist) {
+        qc::QuantumComputation qc{};
+        loadQC(qc, circ);
+        return na::CodeGenerator::generate(
+                   qc, result, maxHOffset, maxVOffset, minEntanglingY,
+                   maxEntanglingY, minAtomDist, noInteractionRadius, zoneDist)
+            .toString();
+      },
+      "circ"_a, "result"_a, "max_h_offset"_a, "max_v_offset"_a,
+      "min_entangling_y"_a, "max_entangling_y"_a, "min_atom_dist"_a = 1,
+      "no_interaction_radius"_a = 10, "zone_dist"_a = 24, R"(
+Generate code for the given circuit using the solver's result. Some parameters
+of the abstraction from the 2D grid used for the solver must be provided again.
+
+.. warning::
+    The result is only consistent, when the same parameters corresponding to
+    them used for the solver.
```

Comment:
Just stumbled upon this while trying to understand where things could be going wrong with the bindings:
Can't you define that error out of existence by adapting this function to take the solver object as a parameter instead of just the result?

Reply:
Even that could go wrong, when the user provides another solver than the one that was to retrieve the result. I decided to respective information to the `Result` class.